File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        
        //Use Map
        unordered_map<char, int> m;
        
        for(int i=0; i<s.size(); i++)
        {
            m[s[i]]++;
        }
        
        for(int i=0; i<s.size(); i++)
        {
            if(m[s[i]] == 1) return i;
        }
        
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int firstUniqChar(String s) {
        // Use Map
        Map<Character, Integer> charCount = new HashMap<>();
        
        for (int i = 0; i < s.length(); i++) {
            charCount.put(s.charAt(i), charCount.getOrDefault(s.charAt(i), 0) + 1);
        }
        
        for (int i = 0; i < s.length(); i++) {
            if (charCount.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        
        return -1;
    }
}
